ipfsapi/client.py

```diff
@@ -743,6 +743,42 @@ def resolve(self, name, recursive=False, **kwargs):
         args = (name,)
         return self._client.request('/resolve', args, decoder='json', **kwargs)
 
+    def key_gen(self, keyName, type='rsa', size='2048', **kwargs):
+        """Adds a new public key that can be used for name_publish.
+
+        .. code-block:: python
+
+            >>> c.key_gen('exampleKeyName')
+            {'Name': 'exampleKeyName',
+             'Id': 'QmQLaT5ZrCfSkXTH6rUKtVidcxj8jrW3X2h75Lug1AV7g8'}
+
+        Parameters
+        ----------
+        keyName : str
+            Name of the new Key to be generated. Used to reference the Keys.
+        type : str
+            Type of key to generate
```

Comment:
Please list the currently possible key types (rsa, ed25519) here so that they appear in the API documentation